### PR TITLE
src/components/form: add `FormFieldCompanyAddress` to challenge registration

### PR DESCRIPTION
**Pull request by maceksimon**
Add `FormFieldCompanyAddress` component (similar to `FormFieldCompany` but without select lookup).

* Add translations
* Add tests

Rename type `FormAddCompanyFields` to `FormCompanyFields` as it should label general Company model (within forms) not only the case of adding a new company.

### CODE REVIEW

**Comment by tmszi** [comment]
Could you resolve conflicts, please?

File: src/i18n/en.toml [hunk 1]
```diff
@@ -95,9 +97,14 @@ titleParticipation = "With whom will you participate in the challenge?"
 hintPariticipation = "If you plan to compete alone, choose a group from which someone could possibly join your team later."
 
 [form.company]
+buttonAddAddress = "Add new"
+buttonAddBranch = "Add branch"
+buttonAddCompany = "Add company"
+hintAddress = "We will deliver a starter pack for your team to this address."
+labelAddress = "Company or branch address"
```
Comment (tmszi) [inline src/i18n/en.toml:104]: Same as my comment above.

File: src/i18n/en.toml [hunk 2]
```diff
@@ -95,9 +97,14 @@ titleParticipation = "With whom will you participate in the challenge?"
 hintPariticipation = "If you plan to compete alone, choose a group from which someone could possibly join your team later."
 
 [form.company]
+buttonAddAddress = "Add new"
+buttonAddBranch = "Add branch"
```
[the remark this answers is not in the document]
Comment (maceksimon) [reply src/i18n/en.toml]: Updated translation terms to "subdivision".

File: src/i18n/en.toml (hunk 1 above)
Comment (maceksimon) [reply src/i18n/en.toml]: Updated translation terms to "subdivision".

**Pull request merged**